pinot-common/src/main/java/org/apache/pinot/common/request/context/RequestContextUtils.java

```diff
@@ -119,7 +119,11 @@ public static FilterContext getFilter(Expression thriftExpression) {
         return new FilterContext(FilterContext.Type.PREDICATE, null,
             new EqPredicate(getExpression(thriftExpression), getStringValue(RequestUtils.getLiteralExpression(true))));
       case LITERAL:
-        // TODO: Handle literals.
+        // Skip "WHERE true"
```

Comment:
it should be eliminated by filter optimization. instead of keeping the where true

Reply:
It's actually not a filter, but the join condition. I will take a deeper look.

Reply:
oh.... shoot. that's unfortunate. for now a temp solution is to skip literal WHERE "true" . please still keep the TODO for handle literal since it is not actually handled

Reply:
Added a PinotLiteralFilterRule to handle this in optimizer.